source/agora/network/Manager.d

```diff
@@ -205,13 +207,6 @@ public class NetworkManager
                 }
                 catch (Exception ex)
                 {
-                    if (auto http = cast(HTTPStatusException)ex)
-                    {
-                        // 404 => API not implemented, it's a FullNode
-                        if (http.status == 404)
-                            break;
-                    }
-
```

Comment:
Actually since I did that change, I had to change `getPublicKey` to `handshake`, otherwise the full node ended up being banned,